indexing-service/src/main/java/io/druid/indexing/worker/executor/ExecutorLifecycle.java

```diff
@@ -47,37 +54,41 @@
 {
   private static final EmittingLogger log = new EmittingLogger(ExecutorLifecycle.class);
 
-  private final ExecutorLifecycleConfig config;
+  private final ExecutorLifecycleConfig taskExecutorConfig;
+  private final TaskConfig taskConfig;
   private final TaskActionClientFactory taskActionClientFactory;
-  private final TaskRunner taskRunner;
+  private final ThreadPoolTaskRunner taskRunner;
   private final ObjectMapper jsonMapper;
 
   private final ExecutorService parentMonitorExec = Execs.singleThreaded("parent-monitor-%d");
 
+  private volatile Task task = null;
   private volatile ListenableFuture<TaskStatus> statusFuture = null;
+  private volatile FileChannel taskLockChannel;
+  private volatile FileLock taskLockFileLock;
 
   @Inject
   public ExecutorLifecycle(
-      ExecutorLifecycleConfig config,
+      ExecutorLifecycleConfig taskExecutorConfig,
+      TaskConfig taskConfig,
       TaskActionClientFactory taskActionClientFactory,
-      TaskRunner taskRunner,
+      ThreadPoolTaskRunner taskRunner,
```

Comment:
ah, nothing, will change back. At one point in this patch "stop()" was not part of TaskRunner